src/app/plugins/marketplace.plugin.ts

```diff
@@ -77,32 +78,62 @@ export class MarketPlacePlugin extends Plugin {
     let i: number;
     let last_id = message.id;
     const itemsForSale: String[] = [];
+    const buffer: Promise<String>[] = [];
 
     for (i = 0; i < limitParam / 100; i++) {
       const config = { limit: 100, before: last_id };
       const batch = await message.channel.fetchMessages(config);
+      //Make sure there are messages
+      if (!batch.array().length) {
+        continue;
+      }
+      last_id = batch.last().id;
+
+      const calls = batch.filter((msg) => this._startsWithPrefix(msg)); //Filter out non !market adds
+      const parsed = calls.map((msg) => this._resolveToListing(msg)); //Turn them into listings
+      parsed.forEach((e) => buffer.push(e));
+    }
 
-      batch.forEach((msg) => {
-        const { content } = msg;
-
-        const startsWithPrefix = [this._NEW_LISTING_PREFIX, this._NEW_ALIAS_PREFIX]
-          .map((e) => content.indexOf(e))
-          .some((e) => e === 1);
-
-        // Make sure nothing is before the prefix
-        if (!startsWithPrefix) {
-          return;
-        }
-
-        let [, item] = content.split('add');
-        if (item?.length) {
-          const user = msg.author;
-          item += '\n' + user; //adds user to end of listing.
-          itemsForSale.push(item);
-        }
-        last_id = msg.id; //last_id will end up as the last message's id
+    await Promise.all(buffer).then((items) => {
+      items.forEach((msg) => {
+        itemsForSale.push(msg);
       });
-    }
+    });
     return itemsForSale;
   }
+
+  private async _resolveToListing(msg: IMessage) {
+    const { content } = msg;
+    let [, item] = content.split('add');
+
+    if (!item?.length) {
+      return '';
+      /*The messages are already filtered before this function is called
+      So this should theoretically never be true*/
+    }
+
+    const user = msg.author;
+    item += '\n' + user; //adds user to end of listing.
```

Comment:
do you have prettier installed? `//<space>comment`

Reply:
Yeah, i just formatted with prettier and nothing changed